File: 1st_week/01_01_find_max_occurred_alphabet.py
def find_max_occurred_alphabet(string):
    alphabet_occurrence_array = [0] * 26

    for char in string:
        if not char.isalpha():
            continue

        alphabet_occurrence_array[ord(char) - ord('a')] += 1

    maxIndex = 0;
    maxCount = 0;
    for i, count  in enumerate(alphabet_occurrence_array):
        if count <= maxCount:
            continue

        maxCount = count
        maxIndex = i

    return chr(maxIndex + ord('a'))
ㅁ

result = find_max_occurred_alphabet
print("정답 = i 현재 풀이 값 =", result("hello my name is dingcodingco"))
print("정답 = e 현재 풀이 값 =", result("we love algorithm"))
print("정답 = b 현재 풀이 값 =", result("best of best youtube"))

File: 1st_week/01_01_find_max_occurred_alphabet.js
function findMaxOccurredAlphabet (str) {
    const alphabets = Array.from({length: 26 }, () => 0);
    const aAscii = "a".charCodeAt();
    
    for (const char of str) {
        if (char === " ") {
            continue;
        }

        const index = char.charCodeAt() - aAscii;
        alphabets[index] += 1;
    }

    let maxCount = 0;
    let maxIndex = 0;
    for (const i in alphabets) {
        if (alphabets[i] > maxCount) {
            maxIndex = i;
            maxCount = alphabets[i];
        }
    }

    return String.fromCharCode(Number(maxIndex) + aAscii);
}

console.log("정답 = i 현재 풀이 값 = " + findMaxOccurredAlphabet("hello my name is dingcodingco"));
console.log("정답 = e 현재 풀이 값 = " + findMaxOccurredAlphabet("we love algorithm"));
console.log("정답 = b 현재 풀이 값 = " + findMaxOccurredAlphabet("best of best youtube"));